src/clang.rs

```diff
@@ -287,11 +287,15 @@ impl Cursor {
 
     /// Given that this cursor points to a template specialization, get a cursor
     /// pointing to the template definition that is being specialized.
-    pub fn specialized(&self) -> Cursor {
+    pub fn specialized(&self) -> Option<Cursor> {
+        if !self.is_valid() {
+            return None;
```

Comment:
Oops, I misread the issue description. I thought the concern was that the *existing* `Cursor` was valid, not that the *resulting* one was valid. Thanks for the correction!